content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,126 @@
++++
+path = "inside-rust/9999/12/31/redesigning-the-initial-bootstrap-sequence"
+title = "Redesigning the Initial Bootstrap Sequence"
+authors = ["Jieyou Xu"]
+
+[extra]
+team = "the Bootstrap team"
+team_url = "https://www.rust-lang.org/governance/teams/infra#team-bootstrap"
++++
+
+This blog post accompanies an [upcoming major change to the `bootstrap` build system][stage0-redesign-pr] (see also [Major Change Proposal 619][redesign-stage0-mcp]), which is expected to affect contributors who works on the `rustc` compiler, the standard library, tools, and the distributors of rust toolchains.
+
+# TL;DR: What is being changed?
+
+We are [redesigning the stage 0 bootstrap sequence][stage0-redesign-pr] so that the standard library will now only support one compiler version. Building the stage 1 compiler will no longer involve building the in-tree standard library. Instead, the pre-built beta standard library will be used.
+
+The following section is a quick primer on the concept of bootstrapping and the terminology we use in this blog post.
+
+# A quick primer on bootstrapping and terminology used in this blog post {#preliminary}
+
+This section is intended to explain some basic bootstrapping concepts to make it easier for contributors to understand the upcoming bootstrap changes. You can [skip this section](#motivation) if you are already familiar with [the `bootstrap` build system][bootstrap] itself.
+
+## What is bootstrapping? {#intro-to-bootstrapping}
+
+[*Bootstrapping*][compiler-bootstrapping] is the process of producing a newer version of a compiler with an older version of itself. Usually, bootstrap processes involve the concept of *stages*, where a compiler and associated artifacts from a stage is used to build the compiler of the next stage. For example, a stage 2 compiler is built from a stage 1 compiler and associated stage 1 artifacts. The stage number is named after the compiler in question.
+
+## Bootstrapping `rustc` {#bootstrapping-rust}
+
+C compilers like [`gcc`][gcc-bootstrap] or [`clang`][clang-bootstrap] do not build their standard libraries from source, but instead links against the same specified standard library across all bootstrap stages. Typically, this specified standard library is the system standard library.[^adapted]
```

Comment:
```suggestion
Building a C++ compiler toolchain like [`gcc`][gcc-bootstrap] or [`clang`][clang-bootstrap] doesn't usually involve building their standard libraries from source, but instead will link against a prebuilt standard library across all bootstrap stages. Typically, this specified standard library is the system standard library.[^adapted]
```

I don't think C has much of a standard library, at least not with the compiler? C++ seems like the closer example.

It also seems key to note that the library *is* built from source, it's just not part of the build process for the compiler...

Reply:
Dropped C, stick with C++ and fixed wording as suggested (round 6).